library/core/src/slice/iter/macros.rs

```diff
@@ -4,9 +4,11 @@
 macro_rules! is_empty {
     // The way we encode the length of a ZST iterator, this works both for ZST
     // and non-ZST.
-    ($self: ident) => {
-        $self.ptr.as_ptr() as *const T == $self.end
-    };
+    ($self: ident) => {{
+        #![allow(unused_unsafe)] // we're sometimes used within an unsafe block
+        // SAFETY: FIXME
+        unsafe { ($self.ptr.as_ptr() as *const u8).offset_from($self.end as *const u8) == 0 }
+    }};
```

Comment:
I will still need to make sure these are safe / Non UB.

If anyone has any Ideas for how to do it better please write a sugestion.

Reply:
Currently not possible without UB, see #92512